service/src/main/java/com/fecklessweasel/service/DepartmentServlet.java

```diff
@@ -0,0 +1,54 @@
+package com.fecklessweasel.service;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import com.fecklessweasel.service.datatier.SQLSource;
+import com.fecklessweasel.service.datatier.SQLInteractionInterface;
+import com.fecklessweasel.service.objectmodel.*;
+import com.fecklessweasel.service.*;
+
+@WebServlet("/servlet/department")
+public final class DepartmentServlet extends HttpServlet {
+
+    /**
+     * Handles post requests to this end point. Performs the creation
+     * of departments given a set of parameters.
+     * **/
+    protected int uniID;
+    protected String deptName;
+    protected String acronym;
+    protected int ID;
+    @Override
+    protected void doPost(final HttpServletRequest request,
+                          final HttpServletResponse response)
+            throws ServletException, IOException {
+
+        SQLSource.interact(new SQLInteractionInterface<Integer>() {
+                               @Override
+                               public Integer run(Connection connection)
+                                       throws ServiceException, SQLException {
+                                   uniID = Integer.parseInt(request.getParameter("university"));
```

Comment:
You are doing no error checking here. If university is null or not an int we will get a NullPointer, ArgumentNull, or NumberFormatException that will crash the page for the user.